### PR TITLE
Virt: Update DPDK cluster configuration instructions

### PR DESCRIPTION
**Pull request by orelmisan**
Version(s):

4.13

Issue:

Link to docs preview:

https://59891--docspreview.netlify.app/openshift-enterprise/latest/virt/virtual_machines/vm_networking/virt-attaching-vm-to-sriov-network.html#virt-configuring-cluster-dpdk_virt-attaching-vm-to-sriov-network

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/virt-configuring-cluster-dpdk.adoc [hunk 1]
```diff
@@ -63,24 +61,46 @@ metadata:
...
-  nodeSelector:
-    node-role.kubernetes.io/worker-dpdk: "" 
 ----
 +
 [NOTE]
 ====
 The compute nodes automatically restart after you apply the `MachineConfigPool` and `PerformanceProfile` manifests.
 ====
 
+. Get the generated runtimeClass from the PerformanceProfile's status:
```
Comment (sjhala-ccs) [inline modules/virt-configuring-cluster-dpdk.adoc:85]:
```suggestion
. Retrieve the name of the generated `RuntimeClass` resource from the `status.runtimeClass` field of the `PerformanceProfile` object :
```

Comment (sjhala-ccs) [inline modules/virt-configuring-cluster-dpdk.adoc:85]:
```suggestion
. Retrieve the name of the generated `RuntimeClass` resource from the `status.runtimeClass` field of the `PerformanceProfile` object :
```

File: modules/virt-configuring-cluster-dpdk.adoc [hunk 2]
```diff
@@ -63,24 +61,46 @@ metadata:
...
+[source,terminal]
```
Comment (sjhala-ccs) [inline modules/virt-configuring-cluster-dpdk.adoc:86]: Fix indentation

```suggestion
+
[source,terminal]
```

File: modules/virt-configuring-cluster-dpdk.adoc [hunk 3]
```diff
@@ -63,24 +61,46 @@ metadata:
...
+----
+$ oc get performanceprofiles.performance.openshift.io profile-1 -o=jsonpath='{.status.runtimeClass}{"\n"}'
+----
+
+. Set the container runtime configuration for the `virt-launcher` pod by adding the following annotation to the `HyperConverged` custom resource (CR):
++
+[source,terminal]
+----
+$ oc annotate --overwrite -n openshift-cnv hco kubevirt-hyperconverged \
+    kubevirt.kubevirt.io/jsonpatch='[{"op": "add", "path": "/spec/configuration/defaultRuntimeClass", "value": <runtimeclass-name>}]'
```
Comment (sjhala-ccs) [inline modules/virt-configuring-cluster-dpdk.adoc:96]:
```suggestion
    kubevirt.kubevirt.io/jsonpatch='[{"op": "add", "path": "/spec/configuration/defaultRuntimeClass", "value": <runtimeclass_name>}]'
```

File: modules/virt-configuring-cluster-dpdk.adoc [hunk 4]
```diff
@@ -63,24 +61,46 @@ metadata:
...
 The compute nodes automatically restart after you apply the `MachineConfigPool` and `PerformanceProfile` manifests.
 ====
 
+. Get the generated runtimeClass from the PerformanceProfile's status:
+[source,terminal]
+----
+$ oc get performanceprofiles.performance.openshift.io profile-1 -o=jsonpath='{.status.runtimeClass}{"\n"}'
+----
+
+. Set the container runtime configuration for the `virt-launcher` pod by adding the following annotation to the `HyperConverged` custom resource (CR):
```
Comment (sjhala-ccs) [inline modules/virt-configuring-cluster-dpdk.adoc:91]:
```suggestion
. Set the container runtime configuration for the `virt-launcher` pod, specifying the`RuntimeClass` name:
```

**Review by sjhala-ccs — COMMENTED** [review]
Thanks @orelmisan. A formatting fix and a few other suggestions based on our style guidelines.

File: modules/virt-configuring-cluster-dpdk.adoc (hunk 1 above)
Comment (orelmisan) [reply modules/virt-configuring-cluster-dpdk.adoc]: Done.

Comment (orelmisan) [reply modules/virt-configuring-cluster-dpdk.adoc]: This seems like a duplicate of the previous comment.

File: modules/virt-configuring-cluster-dpdk.adoc (hunk 2 above)
Comment (orelmisan) [reply modules/virt-configuring-cluster-dpdk.adoc:87]: Done.

**Comment by orelmisan** [comment]
Thank you @sjhala-ccs for your help with this PR.

File: modules/virt-configuring-cluster-dpdk.adoc (hunk 4 above)
Comment (sjhala-ccs) [reply modules/virt-configuring-cluster-dpdk.adoc:92]: I realize that you only moved this step; my suggestion was more to tie it to the previous step and highlight that the `RuntimeClass` name was a user-supplied value.

**Comment by yossisegev** [comment]
LGTM (CNV network QE)

File: modules/virt-configuring-cluster-dpdk.adoc (hunk 4 above)
Comment (phoracek) [reply modules/virt-configuring-cluster-dpdk.adoc:92]: What about merging the two with:

```
Set the earlier obtained `RuntimeClass` name as the default container runtime class for the `virt-launcher` pods by adding the following annotation to the `HyperConverged` custom resource (CR):
```

File: modules/virt-configuring-cluster-dpdk.adoc [hunk 5]
```diff
@@ -63,24 +61,47 @@ metadata:
...
++
+[source,terminal]
+----
+$ oc annotate --overwrite -n openshift-cnv hco kubevirt-hyperconverged \
+    kubevirt.kubevirt.io/jsonpatch='[{"op": "add", "path": "/spec/configuration/defaultRuntimeClass", "value": <runtimeclass_name>}]'
+----
++
+[NOTE]
+====
+Adding the annotation to the `HyperConverged` CR changes a global setting that affects all VMs that are created after the annotation is applied.
```
Comment (phoracek) [inline modules/virt-configuring-cluster-dpdk.adoc:102]:
```suggestion
Adding the annotation to the `HyperConverged` CR changes a global setting that affects all VMs that are created after the annotation is applied. Setting this annotation breaches support of given OpenShift Virtualization instance and should be used only on test clusters. For best performance, apply for a support exception.
```

**Review by phoracek — CHANGES_REQUESTED** [review]

File: modules/virt-configuring-cluster-dpdk.adoc (hunk 5 above)
Comment (orelmisan) [reply modules/virt-configuring-cluster-dpdk.adoc:102]: Done.

File: modules/virt-configuring-cluster-dpdk.adoc (hunk 4 above)
Comment (orelmisan) [reply modules/virt-configuring-cluster-dpdk.adoc]: Changed the wording according to @phoracek's suggestion.

**Review by sjhala-ccs — COMMENTED** [review]
@orelmisan One small change, otherwise LGTM. You will need to squash your commits and then I can merge.

**Comment by orelmisan** [comment]
> @orelmisan One small change, otherwise LGTM. You will need to squash your commits and then I can merge.

Done.

**Pull request merged**